service/src/test/java/org/cbioportal/service/impl/GeneServiceImplTest.java

```diff
@@ -0,0 +1,169 @@
+    }
+
+    @Test
+    public void getAliasesOfGeneByEntrezGeneId() throws Exception {
+
+        List<String> expectedAliases = new ArrayList<>();
+        expectedAliases.add("alias");
+        Mockito.when(geneRepository.getAliasesOfGeneByEntrezGeneId(ENTREZ_GENE_ID)).thenReturn(expectedAliases);
+        List<String> result = geneService.getAliasesOfGene(ENTREZ_GENE_ID.toString());
+
+        Assert.assertEquals(expectedAliases, result);
+    }
+
+    @Test
+    public void getAliasesOfGeneByHugoGeneSymbol() throws Exception {
+
+        List<String> expectedAliases = new ArrayList<>();
+        expectedAliases.add("alias");
+        Mockito.when(geneRepository.getAliasesOfGeneByHugoGeneSymbol(HUGO_GENE_SYMBOL)).thenReturn(expectedAliases);
+        List<String> result = geneService.getAliasesOfGene(HUGO_GENE_SYMBOL);
+
+        Assert.assertEquals(expectedAliases, result);
+    }
+
+    @Test
+    public void fetchGenes() throws Exception {
+
+        List<Gene> expectedGeneList1 = new ArrayList<>();
+        Gene gene1 = new Gene();
+        gene1.setCytoband("12q13.13");
+        expectedGeneList1.add(gene1);
+        List<Gene> expectedGeneList2 = new ArrayList<>();
+        Gene gene2 = new Gene();
+        expectedGeneList2.add(gene2);
+
+        List<String> geneIds = new ArrayList<>();
+        geneIds.add(ENTREZ_GENE_ID.toString());
+        geneIds.add(HUGO_GENE_SYMBOL);
+
+        Mockito.when(geneRepository.fetchGenesByEntrezGeneIds(Arrays.asList(ENTREZ_GENE_ID), PROJECTION))
+                .thenReturn(expectedGeneList1);
+
+        Mockito.when(geneRepository.fetchGenesByHugoGeneSymbols(Arrays.asList(HUGO_GENE_SYMBOL), PROJECTION))
+                .thenReturn(expectedGeneList2);
+
+        List<Gene> result = geneService.fetchGenes(geneIds, PROJECTION);
+
+        Assert.assertEquals(2, result.size());
+        Assert.assertEquals(gene1, result.get(0));
+        Assert.assertEquals(gene2, result.get(1));
+        Assert.assertEquals("12", result.get(0).getChromosome());
+        Assert.assertNull(result.get(1).getChromosome());
+    }
+
+    @Test
+    public void fetchMetaGenes() throws Exception {
+
+        BaseMeta expectedBaseMeta1 = new BaseMeta();
+        expectedBaseMeta1.setTotalCount(1);
+        BaseMeta expectedBaseMeta2 = new BaseMeta();
+        expectedBaseMeta2.setTotalCount(1);
+
+        List<String> geneIds = new ArrayList<>();
+        geneIds.add(ENTREZ_GENE_ID.toString());
+        geneIds.add(HUGO_GENE_SYMBOL);
+
+        Mockito.when(geneRepository.fetchMetaGenesByEntrezGeneIds(Arrays.asList(ENTREZ_GENE_ID)))
+                .thenReturn(expectedBaseMeta1);
+
+        Mockito.when(geneRepository.fetchMetaGenesByHugoGeneSymbols(Arrays.asList(HUGO_GENE_SYMBOL)))
+                .thenReturn(expectedBaseMeta2);
+
+        BaseMeta result = geneService.fetchMetaGenes(geneIds);
+
+        Assert.assertEquals((Integer) 2, result.getTotalCount());
+    }
+}
```

Comment:
Maybe the "empty list" case is prevented because of a required argument at the web layer. Still can't hurt to have these tests eventually.